kubernetes-archv3/dev-v3-secret.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-dev-v3-env
+  labels:
+    app: dev-v3   
+type: Opaque
+data: # data keys encoded by running  `echo -n 'key' | base64` decoded by echo -n 'QWxhZGRpbjpvcGVuIHNlc2FtZQ==' | base64 --decode
+    COUCHDB_PASSWORD: "b3BlbnNlc2FtZQ=="
+    COUCHDB_USER: "Y2h0"
+    API_SVC_NAME: "ZGV2LXYzLW1lZGljLWFwaQ=="
+    COUCHDB_SERVICE_NAME: "ZGV2LXYzLWNvdWNoZGI="
+    COUCH_NODE_NAME: "Y291Y2hkYkAxMjcuMC4wLjE="
+    HAPROXY_SVC: "ZGV2LXYzLWhhcHJveHk="
+    NODE_ENV: "cHJvZHVjdGlvbg=="
+    HORTI_BOOTSTRAP_VERSION: "My4xMC4w"
+    COUCHDB_HOST: "ZGV2LXYzLWNvdWNoZGI="
```

Comment:
Almost none of these need to be "encypted", and of course the password isn't really encrypted. I understand this is intended for local dev and not an internet-exposed service, but how about leaving all of these as plain text and having the `COUCHDB_PASSWORD` read from an environment variable so it doesn't have to be committed to this file?

Reply:
The "Secrets" resource in k8s needs to be base64 encoded (not encrypted). These secrets are environment variables that get injected upon container creation. It's essentially plain-text, we don't check these into git for real projects. 

In the deployment yaml you can find an env section underneath the container declaration that tells k8s to pull the variable values from this secrets resource.

Reply:
> The "Secrets" resource in k8s needs to be base64 encoded (not encrypted). These secrets are environment variables that get injected upon container creation. It's essentially plain-text, we don't check these into git for real projects.
> 
> In the deployment yaml you can find an env section underneath the container spec that tells k8s to pull the variable values from this secrets resource.

Ah, ok. Thanks for the education @Hareet.

Reply:
Before committing these, I thought of removing them from the secrets file but then I thought that someone new to k8s would struggle knowing how to out these in the secrets,yml and how to do base64 encoding., So I put these as samples to help them get started real quick. As a trivial note , one of these has a joke if you decode it ;). Maybe we should put a disclaimer and a warning for people who intend to use these files in production.